SensusService/Protocol.cs

```diff
@@ -376,6 +377,10 @@ public static void RunUnitTestingProtocol(Stream unitTestingProtocolFile)
         private string _name;
         private List<Probe> _probes;
         private bool _running;
+        private bool _scheduledToStart;
+        private bool _scheduledToStop;
+        private string _scheduledStartCallbackId;
+        private string _scheduledStopCallbackId;
```

Comment:
Can't you get rid of the bool properties and interpret a null string as unscheduled?